tfjs-backend-webgpu/src/kernels/Tile.ts

```diff
@@ -0,0 +1,54 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =============================================================================
+ */
+
+import {buffer, KernelConfig, KernelFunc, TensorInfo, Tile, TileAttrs, TileInputs, TypedArray, util} from '@tensorflow/tfjs-core';
+
+import {WebGPUBackend} from '../backend_webgpu';
+import {tileImplCPU} from '../kernel_utils/shared';
+import {TileProgram} from './tile_webgpu';
+
+export function tile(
+    params: {inputs: TileInputs, backend: WebGPUBackend, attrs: TileAttrs}):
+    TensorInfo {
+  const {inputs, backend, attrs} = params;
+  const {x} = inputs;
+  const {reps} = attrs;
+
+  // tile gpu program cannot handle rank >= 5 case.
+  if (x.dtype === 'string' || x.shape.length >= 5) {
```

Comment:
Use `if (backend.shouldExecuteOnCPU(x) || x.dtype === 'string')` ? backend.shouldExecuteOnCPU(x) will include `x.shape.length >= 5`. But if it can't be run on cpu, it can't be run on gpu either. So I think using backend.shouldExecuteOnCPU(x) is _ok._

Reply:
x.shape.length >= 5 is not included in shouldExecuteOnCPU(util.sizeFromShape(input.shape) < sizeThreshold). BTW, I also add shouldExecuteOnCPU.